libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt

```diff
@@ -95,11 +90,18 @@ class RustMatrixClientFactory @Inject constructor(
         }
     }
 
-    internal fun getBaseClientBuilder(
+    internal suspend fun getBaseClientBuilder(
         sessionPaths: SessionPaths,
         passphrase: String?,
-        slidingSync: ClientBuilderSlidingSync,
+        restore: Boolean,
     ): ClientBuilder {
+        val slidingSync = when {
+            AuthenticationConfig.SLIDING_SYNC_PROXY_URL != null -> ClientBuilderSlidingSync.CustomProxy(AuthenticationConfig.SLIDING_SYNC_PROXY_URL!!)
+            appPreferencesStore.isSimplifiedSlidingSyncEnabledFlow().first() -> ClientBuilderSlidingSync.Simplified
+            restore -> ClientBuilderSlidingSync.Restored
```

Comment:
Should we not checked first if restore is `true`?

Reply:
Ooops, true.

Reply:
https://github.com/element-hq/element-x-android/pull/3417/commits/ae3f555fa081f2a6497185c9eeb9b50d0456c70d